clients/src/main/java/org/apache/kafka/common/config/TopicConfigs.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.common.config;
+
+/**
+ * Keys that can be used to configure a topic.  These keys are useful when creating or reconfiguring a
+ * topic using the AdminClient.
+ *
+ * This is a public API, so we should not remove or alter keys without a discussion and a deprecation period.
+ * Eventually this should replace LogConfig.scala.
+ *
+ * Note that some of these are subtly different from the configuration keys that control
+ * the broker defaults.  For example, to set the default broker cleanup policy, you set log.cleanup.policy,
+ * but to set the per-topic cleanup policy, you set cleanup.policy.
+ */
+public class TopicConfigs {
```

Comment:
Shouldn't this be `TopicConfig` to be consistent with `ConsumerConfig`, `ProducerConfig`, etc.?